components/Header.js

```diff
@@ -1,32 +1,29 @@
-import React, { useEffect, useRef, useState } from "react";
+"use client";
+import React, { useEffect, useState, useRef } from "react";
 import Image from "next/image";
-import styles from "../styles/LoginDropdown.module.css";
-import router from "next/router";
-import { useAuth } from "../context/AuthContext";
-import DropdownCard from "../components/LoginDropdown";
+import Link from "next/link";
+import { RxHamburgerMenu } from "react-icons/rx";
+import DropdownCard from "./LoginDropdown";
+import { AiOutlineSearch } from "react-icons/ai";
+import { SunIcon } from "@heroicons/react/24/solid";
+import logo from "../public/logo.png";
 
-function Header({ navbarBackground }) {
-  // localStorage.setItem("redirectUrl", window.location.href);
-  // const [isLoggedIn, setLoggedIn] = useState(false);
+function Header({ isUserLoggedIn }) {
+  const [isLoggedIn, setLoggedIn] = useState(false);
+  const [isMentorLoggedIn, setMentorLoggedIn] = useState(false);
   const [scrollY, setScrollY] = useState(0);
-  const [navbarAppear, setNavbarAppear] = useState(false);
-  const [loginOption, setLoginOption] = useState(false);
-  const {
-    isMentorLoggedIn,
-    setIsMentorLoggedIn,
-    isUserLoggedIn,
-    setIsUserLoggedIn,
-  } = useAuth();
-  const userData = JSON.parse(localStorage.getItem("userData"));
-  const mentorData = JSON.parse(localStorage.getItem("mentorData"));
```

Comment:
please do not rmv this:
1. useAuth manage the auth state
2. We have stored data in that format in local storage so hence we parse too